components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementAdminService.java

```diff
@@ -157,10 +186,16 @@ public IdentityProvider getIdentityProvider(String federatedIdPName)
      * @return Identity providers array
      * @throws org.wso2.carbon.identity.application.common.IdentityApplicationManagementException
      */
-    public IdentityProvider[] getAllIdentityProviders()
-            throws IdentityApplicationManagementException {
-        applicationMgtService = ApplicationManagementService.getInstance();
-        return applicationMgtService.getAllIdentityProviders(getTenantDomain());
+    public IdentityProvider[] getAllIdentityProviders() throws IdentityApplicationManagementException {
+
+        try {
+            applicationMgtService = ApplicationManagementService.getInstance();
+            return applicationMgtService.getAllIdentityProviders(getTenantDomain());
+        } catch (IdentityApplicationManagementException idpException) {
+            log.error("Error while retrieving all identity providers", idpException);
```

Comment:
since we know the tenantDomain using getTenantDomain() method. Shall we include that in the error log?